plugin/trino-kafka/src/main/java/io/trino/plugin/kafka/encoder/DispatchingRowEncoderFactory.java

```diff
@@ -34,9 +34,9 @@ public DispatchingRowEncoderFactory(Map<String, RowEncoderFactory> factories)
         this.factories = ImmutableMap.copyOf(requireNonNull(factories, "factories is null"));
     }
 
-    public RowEncoder create(ConnectorSession session, String dataFormat, Optional<String> dataSchema, List<EncoderColumnHandle> columnHandles)
+    public RowEncoder create(ConnectorSession session, String dataFormat, Optional<String> dataSchema, List<EncoderColumnHandle> columnHandles, String topic, boolean isKey)
```

Comment:
Should we extract some entity that would have things related to the row?  That would have parameters like: ` String dataFormat, Optional<String> dataSchema, List<EncoderColumnHandle> columnHandles, String topic, boolean isKey`. I don't want to create anything artificial, but something that represents some logical thing.

Also please use an `enum` instead of `boolean`.

Reply:
Updated, lmk what you think when you have a chance. Thanks for all the suggestions!